### DISCUSSION TITLE
Why default setting of istio is not istio-cni but istio-init?

### DISCUSSION THREAD

**Discussion by choigonyok** [istio/istio#48803] [Q&A] [answered]
Hi, I'm studying Istio codebase :)

Istio-init uses linux capabilities to config iptables and this can cause `privilege escalation`, which could be bad for security.

So Istio provides istio-cni.

Is there any reason that istio-cni is not a default option? is there any critical problem at istio-cni?

**Comment by rootsongjc** [comment, marked as answer]
I think it could be because of the following reasons:

1. `istio-init` was the original mode provided by Istio, so it's been used by default; 
1. If you use Istio CNI, you may have [compatibility issues](https://istio.io/latest/docs/setup/additional-setup/cni/#compatibility-with-other-cni-plugins) with other CNI Plugin in Kubernetes 
1. Using `istio-init` by default will allow you to get started with Istio quickly, and save you the trouble of having to maintain a CNI.

These are some of my insights, I hope they are useful to you.
